crates/bevy_input/src/gamepad.rs

```diff
@@ -57,3 +64,230 @@ pub enum GamepadAxisType {
+        self.axis_properties
+            .get(&axis)
+            .unwrap_or(&self.default_axis_property)
+    }
+
+    pub fn get_button_axis_properties(&self, button: GamepadButton) -> &ButtonAxisProperty {
+        self.button_axis_properties
+            .get(&button)
+            .unwrap_or(&self.default_button_axis_property)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ButtonProperty {
+    pub press: f32,
+    pub release: f32,
+}
+
+impl Default for ButtonProperty {
+    fn default() -> Self {
+        ButtonProperty {
+            press: 0.75,
+            release: 0.65,
+        }
+    }
+}
+
+impl ButtonProperty {
+    fn is_pressed(&self, value: f32) -> bool {
+        value >= self.press
+    }
+
+    fn is_released(&self, value: f32) -> bool {
+        value <= self.release
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct AxisProperty {
+    pub positive_high: f32,
+    pub positive_low: f32,
+    pub negative_high: f32,
+    pub negative_low: f32,
+    pub threshold: f32,
+}
+
+impl Default for AxisProperty {
+    fn default() -> Self {
+        AxisProperty {
+            positive_high: 0.95,
+            positive_low: 0.05,
+            negative_high: -0.95,
+            negative_low: -0.05,
+            threshold: 0.01,
+        }
+    }
+}
+
+impl AxisProperty {
+    fn filter(&self, new_value: f32, old_value: Option<f32>) -> f32 {
+        if let Some(old_value) = old_value {
+            if (new_value - old_value).abs() <= self.threshold {
+                return old_value;
+            }
+        }
+        if new_value <= self.positive_low && new_value >= self.negative_low {
+            return 0.0;
+        }
+        if new_value >= self.positive_high {
+            return 1.0;
+        }
+        if new_value <= self.negative_high {
+            return -1.0;
+        }
+        new_value
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ButtonAxisProperty {
+    pub high: f32,
+    pub low: f32,
+    pub threshold: f32,
+}
+
+impl Default for ButtonAxisProperty {
+    fn default() -> Self {
+        ButtonAxisProperty {
+            high: 0.95,
+            low: 0.05,
+            threshold: 0.01,
+        }
+    }
+}
+
+impl ButtonAxisProperty {
+    fn filter(&self, new_value: f32, old_value: Option<f32>) -> f32 {
+        if let Some(old_value) = old_value {
+            if (new_value - old_value).abs() <= self.threshold {
+                return old_value;
+            }
+        }
+        if new_value <= self.low {
+            return 0.0;
+        }
+        if new_value >= self.high {
+            return 1.0;
+        }
+        new_value
+    }
+}
+
+#[derive(Default)]
+pub struct GamepadEventState {
+    gamepad_event_reader: EventReader<GamepadEvent>,
+}
+
+pub fn gamepad_event_system(
+    mut state: Local<GamepadEventState>,
+    mut button: ResMut<Input<GamepadButton>>,
+    mut axis: ResMut<Axis<GamepadAxis>>,
+    mut button_axis: ResMut<Axis<GamepadButton>>,
+    event: Res<Events<GamepadEvent>>,
```

Comment:
nit: lets make this `events` because its a collection

Reply:
Done